src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.AsyncIteratorRewriter.cs

```diff
@@ -23,6 +23,9 @@ private sealed class AsyncIteratorRewriter : AsyncRewriter
             // false if it implements IAsyncEnumerator<T>
             private readonly bool _isEnumerable;
 
+            // We use state=-3 to distinguish the initial state from the running state (-1)
+            private const int InitialState = -3;
```

Comment:
>-3; [](start = 45, length = 3)

The `-3` constant is used twice in this PR. Consider adding a named constant to `StateMachinesState`.